pkg/agent/plugin/workloadattestor/k8s/sigstore/sigstore_test.go

```diff
@@ -0,0 +1,1827 @@
+					bundle: &bundle.RekorBundle{
+						Payload: bundle.RekorPayload{
+							Body:           "ewogICJzcGVjIjogewogICAgInNpZ25hdHVyZSI6IHsKICAgICAgImNvbnRlbnQiOiAiTUVVQ0lRQ3llbThHY3Iwc1BGTVA3ZlRYYXpDTjU3TmNONStNanhKdzlPbzB4MmVNK0FJZ2RnQlA5NkJPMVRlL05kYmpIYlVlYjBCVXllNmRlUmdWdFFFdjVObzVzbUE9IgogICAgfQogIH0KfQ==",
+							LogID:          "samplelogID",
+							IntegratedTime: 12345,
+						},
+					},
+				},
+			},
+			containerID: "333333",
+			want: &SelectorsFromSignatures{
+				Subject:        "spirex@example.com",
+				Content:        "MEUCIQCyem8Gcr0sPFMP7fTXazCN57NcN5+MjxJw9Oo0x2eM+AIgdgBP96BO1Te/NdbjHbUeb0BUye6deRgVtQEv5No5smA=",
+				LogID:          "samplelogID",
+				IntegratedTime: "12345",
+			},
+		},
+		{
+			name: "selector from signature, allowedlist enabled, in allowlist, empty content",
+			fields: fields{
+				allowListEnabled: true,
+				subjectAllowList: map[string]bool{
+					"spirex@example.com": true,
+				},
+			},
+			args: args{
+				signature: signature{
+					payload: []byte(`{"critical": {"identity": {"docker-reference": "docker-registry.com/some/image"},"image": {"docker-manifest-digest": "02c15a8d1735c65bb8ca86c716615d3c0d8beb87dc68ed88bb49192f90b184e2"},"type": "some type"},"optional": {"subject": "spirex@example.com","key2": "value 2","key3": "value 3"}}`),
+					bundle: &bundle.RekorBundle{
+						Payload: bundle.RekorPayload{
+							Body:           "ewogICJzcGVjIjogewogICAgInNpZ25hdHVyZSI6IHsKICAgICAgImNvbnRlbnQiOiAiIgogICAgfQogIH0KfQ==",
+							LogID:          "samplelogID",
+							IntegratedTime: 12345,
+						},
+					},
+				},
+			},
+			containerID: "444444",
+			want: &SelectorsFromSignatures{
+				Subject:        "spirex@example.com",
+				LogID:          "samplelogID",
+				IntegratedTime: "12345"},
+		},
+		{
+			name: "selector from signature, no bundle",
+			fields: fields{
+				allowListEnabled: false,
+				subjectAllowList: nil,
+			},
+			args: args{
+				signature: noBundleSignature{
+					payload: []byte(`{"critical": {"identity": {"docker-reference": "docker-registry.com/some/image"},"image": {"docker-manifest-digest": "02c15a8d1735c65bb8ca86c716615d3c0d8beb87dc68ed88bb49192f90b184e2"},"type": "some type"},"optional": {"subject": "spirex@example.com","key2": "value 2","key3": "value 3"}}`),
+				},
+			},
+			containerID: "555555",
+			want: &SelectorsFromSignatures{
+				Subject: "spirex@example.com",
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			sigstore := &sigstoreImpl{
+				allowListEnabled: tt.fields.allowListEnabled,
+				subjectAllowList: tt.fields.subjectAllowList,
+				logger:           hclog.Default(),
+			}
+			if got := sigstore.SelectorValuesFromSignature(tt.args.signature, tt.containerID); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("sigstoreImpl.SelectorValuesFromSignature() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_getBundleSignatureContent(t *testing.T) {
```

Comment:
private functions must not be tested directly, and use functions that call it instead